spec/requests/workers_controller_spec.rb

```diff
@@ -0,0 +1,140 @@
+    end
+  end
+
+  describe "POST /workers action" do
+    before do
+      Worker.destroy_all
+      login_as(sample_worker_user.email, sample_worker_user.password)
+    end
+
+    it "should create a new worker when all necessary params are supplied" do
+      post workers_path, params: { worker: { user_id: sample_worker_user.id, first_name: "Full", last_name: "Name", worker_city: "Sample City", worker_state:"AA", bio: "Sample Bio"} }
+      
+      expect(response).to have_http_status(302)
+      expect(response).to redirect_to user_path(sample_worker_user.id)
+
+      follow_redirect!
+
+      expect(response.body).to include("Worker Account created successfully!")
+      expect(response.body).to include("Your Worker Account Details")
+    end
+
+    it "should display the 'Create Shift' form if one or more necessary params are missing" do
+      post workers_path, params: { worker: { first_name: nil } }
+      
+      expect(response.body).to include("Something went wrong.")
+      expect(response.body).to include("Create Worker Account")
+    end
+  end
+
+  describe "GET /workers/:id action" do
+    it "should display shifts that the worker has taken if worker views their page" do
+      login_as(sample_worker_user.email, sample_worker_user.password)
+      get worker_path(sample_worker.id)
+
+      expect(response).to have_http_status(200)
+      expect(response.body).to include("All Your Shifts")
+    end
+  end
+
+  describe "GET /workers/:id/edit action" do
+    it "should successfully display an 'Edit Worker' form to the worker" do
+      login_as(sample_worker_user.email, sample_worker_user.password)
+      get edit_worker_path(sample_worker.id)
+
+      expect(response).to have_http_status(200)
+      expect(response.body).to include("Edit Workers Account")
+    end
+
+    it "should display error and redirect to users page if the user is not the worker" do
+      login_as(sample_org_user.email, sample_org_user.password)
+      get edit_worker_path(sample_worker.id)
+
+      expect(response).to redirect_to user_path(sample_org_user.id)
+      expect(response).to have_http_status(302)
+
+      follow_redirect!
+
+      expect(response.body).to include("You do not have authority to access that.")
+      expect(response.body).to include("Your Email:")
+    end
+    
+
+
+    # should I test trying to edit a worker as another worker? 
+  end
+
+  describe "PATCH /workers/:id" do
+    it "should update the worker's information" do
+      login_as(sample_worker_user.email, sample_worker_user.password)
+
+      patch worker_path(sample_worker.id), params: { worker: { first_name:"Aegon" } }
+
+      expect(response).to redirect_to user_path(sample_worker_user.id)
+      expect(response).to have_http_status(302)
+
+      follow_redirect!
+
+      expect(response.body).to include("Worker Account updated successfully!")
+      expect(response.body).to include("Aegon")
+    end
+  end
+
+  describe "PUT /workers/:id" do
```

Comment:
We don't need this test Cam because we do not have a PUT action defined in the controller (as far as I know)

Reply:
Thanks @msarit , have removed